CRM/MembershipExtras/BAO/MembershipPeriod.php

```diff
@@ -247,20 +250,45 @@ public static function updatePeriod($params) {
    * with other active periods within the same membership.
    *
    * @param $periodParams
+   *
    * @return bool
+   * @throws \CRM_Core_Exception
    */
   private static function doesOverlapWithOtherActivePeriods($periodParams) {
-    $membershipPeriod = self::getMembershipPeriodById($periodParams['id']);
+    if (!$periodParams['is_active']) {
```

Comment:
if someone use the API to update the period and only changed let's say the end date, then only the end date will be available as a parameter and in case the new end date overlap with other period then it should error out but since you return false, the new end date for the period will be updated even if it overlap with other periods, it is better to fetch the period details first as I did before instead of relaying at the presence of the parameters.